incubator/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -2083,6 +2123,7 @@ private void openSubscribeStreams(
                 MqttSubscribeStream stream = subscribeStreams.computeIfAbsent(subscribeKey, s ->
                     new MqttSubscribeStream(routedId, key, adminSubscribe));
                 stream.packetId = packetId;
+                value.removeIf(s -> s.reasonCode > GRANTED_QOS_2);
```

Comment:
Why is this needed?

Reply:
We don't want to send back to the subscribe stream that "these are the new subscription states" if we received back non-successful suback for that subscription. But we need to receive back these from the sessionStream in order to delegate the SUBACK. So we're removing the non-successful ones when sending the new subscribe filters.